lib/active_job_status/job_tracker.rb

```diff
@@ -24,18 +25,28 @@ def performing
     end
 
     def completed
+      previous_status = store.fetch(job_id)
       store.write(
         job_id,
         JobStatus::COMPLETED.to_s
       )
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
     end
 
     def deleted
+      previous_status = store.fetch(job_id)
       store.delete(job_id)
+      remove_from_batch if batch_id && previous_status && previous_status != JobStatus::COMPLETED.to_s
     end
 
-    private
+    def remove_from_batch
+      remaining_jobs_key = ["remaining_jobs", batch_id].join(":")
```

Comment:
I think the generation of `remaining_jobs_key` and `batch_for_key` should be refactored in a single place (as class methods in one of these classes) to guarantee any future changes to them will apply to all these calls